qutebrowser/misc/overlap.py

```diff
@@ -0,0 +1,127 @@
+# vim: ft=python fileencoding=utf-8 sts=4 sw=4 et:
+
+# Copyright 2021 Florian Bruhin (The Compiler) <mail@qutebrowser.org>
+#
+# This file is part of qutebrowser.
+#
+# qutebrowser is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# qutebrowser is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with qutebrowser.  If not, see <https://www.gnu.org/licenses/>.
+
+"""Work with overlapping widgets."""
+
+import collections
+import math
+from typing import Dict, Iterable, List, Sequence, Set, Tuple
```

Comment:
Since you seem to use type annotations pretty much consistently in this file, could you also add it to `.mypy.ini` with `disallow_untyped_defs` like some other typed modules are? (Ideally, add it next to the other `qutebrowser.misc` sections)